File: java/socketd/src/main/java/org/noear/socketd/transport/core/CodecWriter.java
package org.noear.socketd.transport.core;

import java.io.IOException;

/**
 * 编解码缓冲写
 *
 * @author noear
 * @since 2.0
 */
public interface CodecWriter {
    /**
     * 推入一组 byte
     */
    void putBytes(byte[] bytes) throws IOException;

    /**
     * 推入 int
     */
    void putInt(int val) throws IOException;

    /**
     * 推入 char
     */
    void putChar(int val) throws IOException;

    /**
     * 冲刷
     */
    void flush() throws IOException;
}

File: swift/socketd/socketd/src/socketd/transport/core/CodecWriter.swift
//
//  BufferWriter.swift
//  socketd
//
//  Created by noear on 2023/12/27.
//

import Foundation

/**
 * 编解码缓冲写
 *
 * @author noear
 * @since 2.0
 */
protocol CodecWriter{
    /**
     * 推入一组 byte
     */
    func putBytes( bytes:[Int8]) ;
    
    /**
     * 推入 int
     */
    func putInt( val:Int32) ;
    
    /**
     * 推入 char
     */
    func putChar( val:Int16) ;
    
    /**
     * 冲刷
     */
    func flush() ;
}
